src/Microsoft.ML.Core/Data/ColumnTypeExtensions.cs

```diff
@@ -21,6 +21,32 @@ internal static class ColumnTypeExtensions
         /// <summary>
         /// Zero return means either it's not a key type or the cardinality is unknown.
         /// </summary>
-        public static int GetKeyCount(this ColumnType columnType) => (columnType as KeyType)?.Count ?? 0;
+        public static ulong GetKeyCount(this ColumnType columnType) => (columnType as KeyType)?.Count ?? 0;
+
+        /// <summary>
+        /// Sometimes it is necessary to cast the Count to an int. This performs overflow check.
+        /// </summary>
+        public static int CheckRangeReturnCount(this ColumnType columnType, IExceptionContext ectx = null)
+        {
+            ulong count = columnType.GetKeyCount();
+            ectx.Assert(count <= int.MaxValue, "KeyType range exceeds int.MaxValue.");
+            return (int)count;
+        }
+    }
+
+    /// <summary>
+    /// Extnesion methods related to the KeyType class.
+    /// </summary>
+    [BestFriend]
+    internal static class KeyTypExtensions
+    {
+        /// <summary>
+        /// Sometimes it is necessary to cast the Count to an int. This performs overflow check.
+        /// </summary>
+        public static int CheckRangeReturnCount(this KeyType key, IExceptionContext ectx = null)
+        {
+            ectx.Assert(key.Count <= int.MaxValue, "KeyType range exceeds int.MaxValue.");
```

Comment:
>range [](start = 60, length = 5)

`Count`, isn't it? Also just for the sake of making me happy maybe use something like `nameof(KeyType) + "." + nameof(KeyType.Count)` instead of `"KeyType range"` just to preserve whatever little is left of my sanity.